pages/_error.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * Typescript class based component for custom-error
+ * @link https://nextjs.org/docs/advanced-features/custom-error-page
+ */
+import React from "react";
+import { NextPage, NextPageContext } from "next";
+import NextError, { ErrorProps } from "next/error";
+import { HttpError } from "@lib/core/error/http";
+import { ErrorPage } from "@components/error/error-page";
+import logger from "@lib/logger";
+import { ErrorWithProps } from "@lib/core/error/error-with-props";
+
+// Adds HttpException to the list of possible error types.
+type AugmentedError = (NonNullable<NextPageContext["err"]> & HttpError) | null;
+type CustomErrorProps = {
+  err?: AugmentedError;
+  message?: string;
+  hasGetInitialPropsRun?: boolean;
+} & Omit<ErrorProps, "err">;
+
+type AugmentedNextPageContext = Omit<NextPageContext, "err"> & {
+  err: AugmentedError;
+};
+
+const log = logger.getChildLogger({ prefix: ["[error]"] });
+
+const CustomError: NextPage<CustomErrorProps> = (props) => {
+  const { statusCode, err, message, hasGetInitialPropsRun } = props;
+
+  if (!hasGetInitialPropsRun && err) {
+    // getInitialProps is not called in case of
+    // https://github.com/vercel/next.js/issues/8592. As a workaround, we pass
+    // err via _app.tsx so it can be captured
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const e =
+      (err as unknown) instanceof Error
+        ? err
+        : ErrorWithProps.createFromProps(err as unknown, "Returned err is not an Error object");
```

Comment:
Why would we want to do this? 

Better:

```ts
export function getErrorFromUnknown(cause: unknown): Error {
  if (cause instanceof Error) {
    return cause;
  }
  if (typeof cause) {
    // @ts-expect-error https://github.com/tc39/proposal-error-cause
    return new Error(cause, { cause });
  }

  return new Error(`Unhandled error from type '${typeof cause}''`);
}
```

export function getErrorFromUnknown(cause: unknown): TRPCError {
  // this should ideally be an `instanceof TRPCError` but for some reason that isn't working
  // ref https://github.com/trpc/trpc/issues/331
  if (cause instanceof Error && cause.name === 'TRPCError') {
    return cause as TRPCError;
  }
  const err = new TRPCError({
    code: 'INTERNAL_SERVER_ERROR',
    cause,
  });

  // take stack trace from cause
  if (cause instanceof Error) {
    err.stack = cause.stack;
  }
  return err;
}
```

then,

```suggestion
    const e = getErrorFromUnknown(err)
```